### PR TITLE
Linq Testing Improvements and Consolidations

### PR DESCRIPTION
**Pull request by jeremydmiller**
Kind of a sideways PR to deal with #567 in preparation for getting serious about working with the Linq improvements

### CODE REVIEW

File: src/Marten.Testing/Linq/Compatibility/Support/TargetSchemaFixture.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Marten.Testing.Linq.Compatibility.Support
+{
+
+    public abstract class TargetSchemaFixture : IDisposable
+    {
+        public readonly Target[] Documents = Target.GenerateRandomData(1000).ToArray();
```
Comment (oskardudycz) [inline src/Marten.Testing/Linq/Compatibility/Support/TargetSchemaFixture.cs]: Sidenote: wouldn't 100 be enough and make tests run faster?

File: src/Marten.Testing/Linq/Compatibility/Support/LinqTestContext.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            while (!path.EndsWith("Marten.Testing"))
+            {
+                path = path.ParentDirectory();
+            }
+
+            var filename = typeof(TSelf).Name + ".cs";
+            var codefile = path.AppendPath("Linq", "Compatibility", filename);
+
+            var list = new List<string>();
+
```
Comment (oskardudycz) [inline src/Marten.Testing/Linq/Compatibility/Support/LinqTestContext.cs]: redundant new lines

File: src/Marten.Testing/Linq/Compatibility/Support/TargetSchemaFixture.cs (hunk 1 above)
Comment (jeremydmiller) [reply src/Marten.Testing/Linq/Compatibility/Support/TargetSchemaFixture.cs]: Dunno, but it's been pretty fast with 1K

**Comment by jeremydmiller** [comment]
@oskardudycz If we need to ignore some cases, I'd break those out into a separate test fixture

**Comment by mysticmind** [comment]
@jeremydmiller sorry about my delayed review on this, the changes look good to me. CI tests are taking between 5-6mins, do we expect the current changes to reduce the time taken for the tests?

**Pull request merged**